typescript/agentkit/src/action-providers/spl/splActionProvider.ts

```diff
@@ -56,12 +57,17 @@ export class SplActionProvider extends ActionProvider<SvmWalletProvider> {
       const { getMint, getAssociatedTokenAddress, getAccount, TokenAccountNotFoundError } =
         await import("@solana/spl-token");
 
-      const mintInfo = await getMint(connection, mintPubkey);
-      const ata = await getAssociatedTokenAddress(mintPubkey, ownerPubkey);
+      let mintInfo: Mint;
+      try {
+        mintInfo = await getMint(connection, mintPubkey);
+      } catch (error) {
+        return `Failed to fetch mint info for mint address ${args.mintAddress}. Error: ${error}`;
+      }
 
       try {
+        const ata = await getAssociatedTokenAddress(mintPubkey, ownerPubkey);
         const account = await getAccount(connection, ata);
-        const balance = Number(account.amount) / Math.pow(10, mintInfo.decimals);
+        const balance = Number(account.amount) / Math.pow(10, mintInfo!.decimals);
```

Comment:
nit: `!` not necessary here
```suggestion
        const balance = Number(account.amount) / Math.pow(10, mintInfo.decimals);
```